ql/pom.xml

```diff
@@ -1072,6 +1072,7 @@
               <artifactSet>
                 <includes>
                   <!-- order is meant to be the same as the ant build -->
+                  <include>org.antlr:antlr-runtime</include>
```

Comment:
I did not figure this out, however we discussed with @abstractdog offline and we agreed that it wasn't on the tez's side, so there were no other options left. A reasonable explanation would be that antlr's classes where shipped by one of the dependencies in Hive, which changed behaviour.

Reply:
I run `for f in *.jar; do echo $f; jar tf $f | grep CommonTree; done` on the lib folder (for both 3.1.2 and master) and I see the following:
**Hive 3.1.2**
```
antlr-runtime-3.5.2.jar
org/antlr/runtime/tree/CommonTree.class
org/antlr/runtime/tree/CommonTreeAdaptor.class
org/antlr/runtime/tree/CommonTreeNodeStream.class
...
hive-druid-handler-3.1.2.jar
org/antlr/runtime/tree/CommonTree.class
org/antlr/runtime/tree/CommonTreeAdaptor.class
org/antlr/runtime/tree/CommonTreeNodeStream.class
org/skife/jdbi/org/antlr/runtime/tree/CommonTree.class
org/skife/jdbi/org/antlr/runtime/tree/CommonTreeAdaptor.class
org/skife/jdbi/org/antlr/runtime/tree/CommonTreeNodeStream.class
```
**master**
```
antlr-runtime-3.5.2.jar
org/antlr/runtime/tree/CommonTree.class
org/antlr/runtime/tree/CommonTreeAdaptor.class
org/antlr/runtime/tree/CommonTreeNodeStream.class
...
hive-druid-handler-4.0.0-SNAPSHOT.jar
org/antlr/runtime/tree/CommonTree.class
org/antlr/runtime/tree/CommonTreeAdaptor.class
org/antlr/runtime/tree/CommonTreeNodeStream.class
org/skife/jdbi/org/antlr/runtime/tree/CommonTree.class
org/skife/jdbi/org/antlr/runtime/tree/CommonTreeAdaptor.class
org/skife/jdbi/org/antlr/runtime/tree/CommonTreeNodeStream.class
...
hive-exec-4.0.0-SNAPSHOT.jar
org/antlr/runtime/tree/CommonTree.class
org/antlr/runtime/tree/CommonTreeAdaptor.class
org/antlr/runtime/tree/CommonTreeNodeStream.class
```

Reply:
Adding/Removing things from the shaded jar can have a big impact on projects using Hive so that's why I am insisting a bit more on this. Have you checked if it is an issue with the classloader (e.g., HIVE-21971)?

Reply:
It's the Tez worker throwing the classnotfound exception, I don't think that Hive plays any role on that. From what you said earlier, it's also clear that nothing has changed on the Hive side regarding the shipping of antlr classes. 

I have checked the jars for tez 0.9.1 and 0.10.0 (in all the locations), none of them has the antlr classes either, so I am a bit puzzled, they were either not needed before, or there is something trickier like class loader issues. I am not sure where to go from here, but I understand your concern.